consul_kv.go

```diff
@@ -0,0 +1,52 @@
+package onlinelab
+
+import (
+	"github.com/hashicorp/consul/api"
+	"github.com/json-iterator/go"
+)
+
+var json = jsoniter.ConfigCompatibleWithStandardLibrary
+var kv *api.KV
+
+func NewConsulConfigStorage(consulAddress string) (*ConsulConfigStorage, error) {
+	//var err error
+	client, err := api.NewClient(&api.Config{Address: consulAddress})
+	if err != nil {
+		return nil, err
+	}
+
+	// Get a handle to the KV API
+	kv = client.KV()
+	return &ConsulConfigStorage{}, nil
+}
+
+type ConsulConfigStorage struct {
+	config Config
+}
+
+func (cs *ConsulConfigStorage) GetConfig(labName string) (Config, error) {
+	pair, _, err := kv.Get(labName, nil)
+	if err != nil {
+		return cs.config, err
```

Comment:
Some comments is required for the trick logic about returning the original config when failing to get config from Consul.